pkg/helpers/helpers.go

```diff
@@ -140,6 +173,14 @@ func UpdateKlusterletStatus(
 		oldStatus := &klusterlet.Status
 
 		newStatus := oldStatus.DeepCopy()
+		// TODO: just for upgrading, we change the condition type from "InvalidRegistrationFeatureGates" to
+		// "ValidRegistrationFeatureGates", need to remove this in 0.11.0
+		if err := RemoveKlusterletConditionFn("InvalidRegistrationFeatureGates",
+			"ValidRegistrationFeatureGates", "ValidWorkFeatureGates")(newStatus); err != nil {
```

Comment:
do we have ValidWorkFeatureGates, ValidRegistrationFeatureGates already? I think we introduce it recently, and we have not released 0.10.0

Reply:
[clusterManager](https://github.com/open-cluster-management-io/registration-operator/blob/v0.9.0/pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go#L141-L142) has ValidWorkFeatureGates, ValidRegistrationFeatureGates in 0.9.0
and [klusterlet](https://github.com/open-cluster-management-io/registration-operator/blob/v0.9.0/pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go#L249) has ValidRegistrationFeatureGates and a wrong InvalidRegistrationFeatureGates, but does not have ValidWorkFeatureGates in 0.9.0. so I will delete the removing of ValidWorkFeatureGates for klusterlet, but keep others.